### PR TITLE
Add type parameter for ReplaceVisitor context

### LINKED ISSUE #1068: Add type parameter for context of ReplaceVisitor
Add type parameter for context of ReplaceVisitor

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1068

### CODE REVIEW

**Comment by timowest** [comment]
@Shredder121 Is this good to merge?

File: querydsl-jpa/src/main/java/com/querydsl/jpa/JPAMapAccessVisitor.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ public JPAMapAccessVisitor(QueryMetadata metadata) {
 
     public Expression<?> visit(Path<?> expr, @Nullable Void context) {
         expr = (Path<?>) super.visit(expr, null);
-        PathMetadata pathMetadata = expr.getMetadata();
+        PathMetadata<?> pathMetadata = expr.getMetadata();
```
Comment (Shredder121) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/JPAMapAccessVisitor.java]: Could you remove this? this clashes with #1145.

**Comment by timowest** [comment]
Done

File: querydsl-jpa/src/main/java/com/querydsl/jpa/JPAMapAccessVisitor.java (hunk 1 above)
Comment (timowest) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/JPAMapAccessVisitor.java]: Done

**Comment by Shredder121** [comment]
It now looks good to me.
I will merge it then.

**Pull request merged**